unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/GodMode/Inspector/EntityListAdapter.cs

```diff
@@ -88,6 +95,32 @@ void SetInfo(DCLBuilderInWorldEntity entityToEdit)
         }
     }
 
+    internal void GetThumbnail(CatalogItem catalogItem)
+    {
+        if (catalogItem == null)
+            return;
+
+        var url = catalogItem.thumbnailURL;
+
+        if (string.IsNullOrEmpty(url))
+            return;
+
+        var newLoadedThumbnailPromise = new AssetPromise_Texture(url);
+        newLoadedThumbnailPromise.OnSuccessEvent += SetThumbnail;
+        newLoadedThumbnailPromise.OnFailEvent += x => { Debug.Log($"Error downloading: {url}"); };
+        AssetPromiseKeeper_Texture.i.Keep(newLoadedThumbnailPromise);
+        AssetPromiseKeeper_Texture.i.Forget(loadedThumbnailPromise);
+        loadedThumbnailPromise = newLoadedThumbnailPromise;
```

Comment:
I think you don't need the `loadedThumbnailPromise` variable. As far as I see, you're setting it a value but never using it.

Reply:
Yes, I am using it just the line before, I am using the `Forget `method to free the promise keeper. This is just in case we start pooling the adapter, this will make sure that we are only using the current promise keeper